### PR TITLE
[Workaround] Disable bwd v1r1 xdlops due to compiler issues

### PR DESCRIPTION
**Pull request by zjing14**
- Disabled bwd v1r1 xdlops due to the issue #456
- Related to SWDEV_251757
- Merge after #466

### CODE REVIEW

**Comment by daniellowell** [comment]
@asleepzzz This PR is a workaround. Please implement a fix as soon as possible.

File: src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp [hunk 1]
```diff
@@ -635,6 +633,14 @@ size_t ConvHipImplicitGemmBwdDataV1R1::GetWorkspaceSize(const ConvolutionContext
 
 bool ConvHipImplicitGemmBwdDataV1R1::IsApplicable(const ConvolutionContext& ctx) const
 {
+/// \todo Fix and remove this workaround.
+/// There are failures with certain configs,
+/// see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/456
+#if 1
```
Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp:639]: Please use something like ```#if WORKAROUND_GITHUB_456``` so it's easy to grep all workaround in the code

Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp:639]: Please use something like ```#if WORKAROUND_GITHUB_456``` so it's easy to grep all workaround in the code

**Review by asroy — CHANGES_REQUESTED** [review]

File: src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp (hunk 1 above)
Comment (asroy) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp]: Actually, please refer to the JIRA ticket instead of github issue.

**Review by atamazov — CHANGES_REQUESTED** [review]
We have to disable the Solver beginning from the specific compiler version (that fail). Please observe compiler version. I recommend borrowing design from here:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/1f319e1b87451fb0b8ed32a6b41118daf6122d4f/src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp#L653-L657

**Comment by asroy** [comment]
> @asleepzzz This PR is a workaround. Please implement a fix as soon as possible.

@daniellowell The failure is due to compiler bug, a JIRA has been created for compiler for several weeks now (SWDEV_251757). MIOpen can only create a workaround for it, and this workaround is to completely disable it.

@asleepzzz didn't work on this kernel, he worked on bwd-data v4r1, which currently is faster than this bwd-data v1r1 kernel in all the test we did. So it's OK to completely disable this v1r1 until compiler fixed the bug.

File: src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp [hunk 2]
```diff
@@ -753,6 +753,11 @@ ConvHipImplicitGemmBwdDataV1R1Xdlops::GetWorkspaceSize(const ConvolutionContext&
 
 bool ConvHipImplicitGemmBwdDataV1R1Xdlops::IsApplicable(const ConvolutionContext& ctx) const
 {
+#if WORKAROUND_SWDEV_251757
+    (void)ctx;
+    if(miopen::HipCompilerVersion() >= external_tool_version_t{3, 7, 0})
+        return false;
+#else
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp:760]: [Notice] BTW now you can replace `#else` with `#endif`, and remove `(void)ctx`  and `#endif` at line 788.

**Review by atamazov — DISMISSED** [review]

File: src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp:760]: ...just like shown at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/457#pullrequestreview-493915098

File: src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp (hunk 1 above)
Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v1r1.cpp]: I believe this is resolved

**Review by atamazov — APPROVED** [review]

**Comment by daniellowell** [comment]
@zjing14 This PR has introduced some major issues.
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/igemm_disable_bwd_v1r1/5/pipeline

**Comment by atamazov** [comment]
> @zjing14 This PR has introduced some major issues.
> http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/igemm_disable_bwd_v1r1/5/pipeline

Well, not introduced but revealed. The issue is in `ConvHipImplicitGemmBwdDataV4R1Xdlops`. It was masked by V1R1 solver, because iGemm algorithm still uses the "Find Only First" logic (see WORKAROUND_SWDEV_227826).

**Comment by atamazov** [comment]
Solver history suggests that @jane-zxy and @shaojiewang can also help with V4R1 xdlops failure. https://github.com/ROCmSoftwarePlatform/MIOpen/commits/develop/src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp

**Comment by zjing14** [comment]
> @zjing14 This PR has introduced some major issues.
> http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/igemm_disable_bwd_v1r1/5/pipeline

This failure is due to improper tuning parameters. @asleepzzz could fix it.

**Comment by atamazov** [comment]
CI checks passed.

**Comment by daniellowell** [comment]
@zjing14  This is still not fixed:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/igemm_disable_bwd_v1r1/9/pipeline

You have something else going on.

**Comment by asroy** [comment]
@asleepzzz @zjing14 
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/466#issuecomment-700074055

**Review by asroy — APPROVED** [review]

**Pull request merged**